roles/openshift_node40/tasks/config.yml

```diff
@@ -47,3 +21,38 @@
     name: container_manage_cgroup
     state: yes
     persistent: yes
+
+- name: Get machine controller daemon image from release image
+  command: "podman run --rm {{ openshift_release_image }} image machine-config-daemon"
+  register: release_image_mcd
+
+- name: Copy bootstrap ignition file locally
+  copy:
+    src: "{{ openshift_ignition_file_path }}"
+    dest: "{{ ign_file }}"
+  when: openshift_ignition_file_path is defined
+
+- name: Fetch bootstrap ignition file locally
+  uri:
+    url: "{{ openshift_bootstrap_endpoint }}"
+    dest: "{{ ign_file }}"
+    validate_certs: false
+  when: openshift_bootstrap_endpoint is defined
+
+- block:
+  - name: Apply ignition manifest
+    command: "podman run {{ podman_mounts }} {{ podman_flags }} {{ mcd_command }}"
+    vars:
+      podman_flags: "--privileged --rm -ti {{ release_image_mcd.stdout }}"
+      podman_mounts: "-v /:/rootfs -v /var/run/dbus:/var/run/dbus -v /run/systemd:/run/systemd"
+      mcd_command: "start --node-name {{ ansible_hostname }} --once-from {{ ign_file }}"
```

Comment:
Why is a --node-name required here? Does this add any risk of the hostname snafus we had in the past?

Reply:
This is a required parameter to MCO currently. It seems to be required to distinct hostname in container from real node's hostname